src/bucket/BucketTests.cpp

```diff
@@ -1109,36 +1109,60 @@ TEST_CASE("bucket apply", "[bucket]")
     REQUIRE(count == 1);
 }
 
-#ifdef USE_POSTGRES
 TEST_CASE("bucket apply bench", "[bucketbench][!hide]")
 {
-    VirtualClock clock;
-    Config cfg(getTestConfig(0, Config::TESTDB_POSTGRESQL));
-    Application::pointer app = createTestApplication(clock, cfg);
-    app->start();
+    auto runtest = [](Config::TestDbMode mode) {
+        VirtualClock clock;
+        Config cfg(getTestConfig(0, mode));
+        Application::pointer app = createTestApplication(clock, cfg);
+        app->start();
 
-    std::vector<LedgerEntry> live(100000);
-    std::vector<LedgerKey> noDead;
+        std::vector<LedgerEntry> live(100000);
+        std::vector<LedgerKey> noDead;
 
-    for (auto& l : live)
-    {
-        l.data.type(ACCOUNT);
-        auto& a = l.data.account();
-        a = LedgerTestUtils::generateValidAccountEntry(5);
-    }
+        for (auto& l : live)
+        {
+            l.data.type(ACCOUNT);
+            auto& a = l.data.account();
+            a = LedgerTestUtils::generateValidAccountEntry(5);
+        }
 
-    std::shared_ptr<Bucket> birth =
-        Bucket::fresh(app->getBucketManager(), live, noDead);
+        std::sort(live.begin(), live.end(),
+                  [](LedgerEntry const& a, LedgerEntry const& b) {
+                      return LedgerEntryIdCmp{}(a.data, b.data);
+                  });
 
-    auto& db = app->getDatabase();
-    auto& sess = db.getSession();
+        LedgerEntry prev;
+        for (auto& l : live)
+        {
+            if (!LedgerEntryIdCmp{}(prev.data, l.data))
+            {
+                auto k = KeyUtils::toStrKey(l.data.account().accountID);
+                CLOG(INFO, "Bucket") << "With duplicate bucket: " << k;
+            }
+            prev = l;
+        }
 
-    CLOG(INFO, "Bucket") << "Applying bucket with " << live.size()
-                         << " live entries";
-    {
-        soci::transaction sqltx(sess);
+        std::shared_ptr<Bucket> birth =
+            Bucket::fresh(app->getBucketManager(), live, noDead);
```

Comment:
If you don't loop to check for duplicate accounts (lines 1135-1144), then there will be no need to `std::sort` the entries in advance (lines 1130-1133) since that is handled as part of `Bucket::fresh`.

Reply:
yes, got rid of that